xclim/indices/run_length.py

```diff
@@ -639,6 +651,85 @@ def get_out(rls):
     return da.copy(data=out.transpose(*da.dims).data)
 
 
+def rle_with_holes(
+    da_start: xr.DataArray,
+    window_start: int,
+    da_stop: xr.DataArray,
+    window_stop: int,
+    index: str = "first",
+    dim: str = "time",
+) -> xr.DataArray:
+    """Generate modified run length function with run sequences that can contain holes.
```

Comment:
What the function does is actually more generic. What would correspond IMO to `rle_with_holes` would be with the additional constraint `da_stop = 1 - da_start`. In this way, it would be very similar to a season, except you allow to set the window of holes (instead of having the same window as the start condition).

Maybe finding a better name and maybe adding a function `rle_with_holes` simply setting `da_stop = 1 - da_start` would be interesting

Reply:
Some input on whether this function does what we want would be good.

Usually, a season is something like:
da = [**1,1,1**, 0,1,**0,0,0**,1...]
We want a given window (say 3) to be true, then, once the opposite condition is found again the season ends.  The season starts at **1 1 1 ** and ends at **0 0 0**. 

You can see this as two run lengths being computed, one that find the sequence **1 1 1**, another that finds **0 0 0**.

I generalized this idea by decoupling these two run lengths searches by defining 
`da_start` and `da_stop`. In the simple example above, `da_stop = 1- da_start`

This can lead to weird situations:
`da_start = [1,1,1,1,1,1,1,1,1,0,0,0,0,0,0,0, .........], window_start = 1`
`da_stop = [0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0, ....], window_stop = 1`
will give: `run_lengths = [2,NaN, 0, ~ len(da_start)]`

The reason is that 
[1,1,1,**1**,1,1,1,1,1,0,0,0,0,0,0,0, .........]
has nothing to stop it once we pass the first and only stop. In this sense, it works much more like a season than a run_length, so maybe a change of name would be warranted.

This weird behaviour will likely not happen when `da_stop` and `da_start`  are related like `da_stop  = 1 - da_start` like in the simpler case.

Another choice I had made above: The run starts backs _after_ we have passed the stop, i.e. we obtain 
`[2,NaN, 0, ~ len(da_start)]`
and not
`[2,NaN, 1, ~ len(da_start) + 1]`

Again, this is a very specific corner case that may not play an important role in many cases, but if you know of any situation where this is important, we actually have an idea of what we want, that would be great.

For "verglas" for instance, the only generalization needed is really having `window_start` and `window_stop` I believe. The fact we have `da_stop = 1 - da_start` will make it so that there is no overlapping ambiguity such as above.